### PR TITLE
Move cloud_cfg datasource into separate module

### PR DESCRIPTION
**Pull request by bfahr**
### All Pull Requests:

Check all that apply:

* [X] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [X] Is this PR an enhancement?

### Complete Description of Additions/Changes:
Establish a new pattern for the implementation of custom datasources using the cloud_cfg datasource

* Move this datasource out of default.py into its own module
* Create tests for this datasource
* Update documentation for all custom datasources

Signed-off-by: Bob Fahr <20520336+bfahr@users.noreply.github.com>

### CODE REVIEW

**Review by lhuett — APPROVED** [review]
Looks good to me

File: insights/specs/datasources/cloud_init.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+class LocalSpecs(Specs):
+    """ Local specs used only by cloud_init datasources """
+
+    cloud_cfg_input = simple_file("/etc/cloud/cloud.cfg")
+    """ Returns the contents of the file ``/etc/cloud/cloud.cfg`` """
+
+
+@datasource(LocalSpecs.cloud_cfg_input, HostContext)
+def cloud_cfg(broker):
```
Comment (huali027) [inline insights/specs/datasources/cloud_init.py:22]: @bfahr, The datasource `cloud_cfg` should be in the class `LocalSpecs`, or else insights.specs.Specs.cloud_cfg can't find the dependency, right?

Comment (bfahr) [reply insights/specs/datasources/cloud_init.py:22]: @huali027 `cloud_cfg` that is defined in default.py is the data that will be collected in the archive.  `cloud_cfg_input` is only collected for the datasource processing, and is not stored in the archive.  Its defined locally in the same module to simplify the implementation.

Comment (huali027) [reply insights/specs/datasources/cloud_init.py:22]: @bfahr, got it, I found "cloud_cfg = cloud_init.cloud_cfg" in default.py, thanks.

**Pull request merged**